### PR TITLE
Add new eras to support different Run 2 configurations

### PR DESCRIPTION
**Pull request by mark-grimes**
This creates new sub-eras to aid with the Run 2 divergence between 25ns, 50ns and HI.  The user will specify one of `Run2_25ns`, `Run2_50ns` or `Run2_HI` (as discussed in the 17th Feb ORP) but these are actually chains of the new modifiers created here.

I also changed the way cmsDriver checks whether one of the eras specified is one of the internal use sub-eras, so that it can warn the user.

**I propose getting rid of the eras `run2`, `bunchspacing25ns` and `bunchspacing50ns`**, but this isn't done in this pull request.  Everything so far configured as `run2` will be changed to `run2_common` which is added here, then `run2` removed.  This will be a big (but simple) change in many packages so I wanted this idea reviewed here first.  The bunchspacing ones will be changed to `run2_25ns_specific` or `run2_50ns_specific` then removed.

Current road map for eras is a bit different to that discussed in the 17th Feb ORP:
- This pull request
- Change all `run2` modifications to `run2_common` and remove `run2` [EDIT implemented in #7841]
- L1 trigger changes (the 1 pull request mentioned in the ORP) [EDIT implemented in #7860]
- potentially another L1T for FastSim pull request, once I get an answer to a question
- Validation, then any pull requests to fix errors found

@fwyzard, you asked to be kept informed
@Dr15Jones, in case you have any comments on the choice of chaining

### CODE REVIEW

**Comment by Dr15Jones** [comment]
seems reasonable to me

**Comment by mark-grimes** [comment]
Pull request #7841 makes the change from `run2` to `run2_common` mentioned above.  I also have the branch to take out the old bunchspacing eras ready to go, but if I put it in now it'll want loads of signatures because it includes the #7841 changes.

File: Configuration/StandardSequences/python/Eras.py [hunk 1]
```diff
@@ -10,4 +10,25 @@ def __init__(self):
...
+        self.run2_common = cms.Modifier()
+        self.run2_25ns_specific = cms.Modifier()
+        self.run2_50ns_specific = cms.Modifier()
+        self.run2_HI_specific = cms.Modifier()
+        self.stage1L1Trigger = cms.Modifier()
+        
+        # These are the eras that the user should specify
+        self.Run2_25ns = cms.ModifierChain( self.run2_common, self.run2_25ns_specific, self.stage1L1Trigger )
+        self.Run2_50ns = cms.ModifierChain( self.run2_common, self.run2_50ns_specific )
+        self.Run2_HI = cms.ModifierChain( self.run2_common, self.run2_HI_specific, self.stage1L1Trigger )
```
Comment (fwyzard) [inline Configuration/StandardSequences/python/Eras.py]: Hi Mark,
as far as I know, the L1 Stage 1 trigger configuration for Heavy Ions is slightly different than that for pp/25ns.
How is that handled here ?

Comment (mark-grimes) [inline Configuration/StandardSequences/python/Eras.py]: I hadn't realised there was a difference, other than the menu and `simCaloStage1Digis.FirmwareVersion` (from looking at the [`customiseSimL1EmulatorForPostLS1_HI `](https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_0_pre7/L1Trigger/L1TCommon/python/customsPostLS1.py#L129) function).  The `stage1L1Trigger` era I planned to implement everything in the [`customiseSimL1EmulatorForStage1`](https://github.com/cms-sw/cmssw/blob/CMSSW_7_4_0_pre7/L1Trigger/L1TCommon/python/customsPostLS1.py#L10) function and then the other changes (menu and firmware) I was going to put in `run2_HI_specific`.  Are there other differences?

Comment (mark-grimes) [inline Configuration/StandardSequences/python/Eras.py]: Here's the branch I have for the L1 customisations.  I thought it was mostly finished and I was planning to submit it today after testing, unless you have any suggestions for changes?

https://github.com/mark-grimes/cmssw/compare/cms-sw:CMSSW_7_4_X...mark-grimes:addL1TEraChanges

Comment (fwyzard) [inline Configuration/StandardSequences/python/Eras.py]: Hi Mark,
those are the only two changes I know of, so if you are going to handle them in `run2_HI_specific` that should be ok.

Comment (fwyzard) [inline Configuration/StandardSequences/python/Eras.py]: Ouch, I hadn't realised that the syntax for adding the era customisations would be so... inelegant :(

**Pull request merged**